### ISSUE TITLE
SSH tunnel using ssh key will not work with Paramiko != 2.7.2

### ISSUE THREAD

**Issue by survivor956** [pahaz/sshtunnel#257]
https://github.com/pahaz/sshtunnel/blob/58a46ef8b66be5eded29cd9bda4dccedb46ec586/sshtunnel.py#L1418

The problem occurs when using ssh key authentication.
```
try:
    self._transport = self._get_transport()
    self._transport.connect(hostkey=self.ssh_host_key,
        username=self.ssh_username,
        password=self.ssh_password)
    if self._transport.is_alive:  # This will return False (same behaviour as paramiko transport is_active)
        return  # No return
except paramiko.AuthenticationException:
    self.logger.debug('Authentication error')
    self._stop_transport()

self.logger.error('Could not open connection to gateway')  # this code is reached
```
When initiating connection, a paramiko transport is created in a separate thread via the method [start_client](https://docs.paramiko.org/en/2.7/api/transport.html#paramiko.transport.Transport.start_client) which will FAIL with any version of paramiko other than **`2.7.2`** the paramiko transport `is_active` method will return `False` so also `is_alive`

Then the program will not return [here](https://github.com/pahaz/sshtunnel/blob/58a46ef8b66be5eded29cd9bda4dccedb46ec586/sshtunnel.py#L1407) it reaches [this error logging](https://github.com/pahaz/sshtunnel/blob/58a46ef8b66be5eded29cd9bda4dccedb46ec586/sshtunnel.py#L1427) all the way the the same check and exception raising [here](https://github.com/pahaz/sshtunnel/blob/58a46ef8b66be5eded29cd9bda4dccedb46ec586/sshtunnel.py#L1330)

Long story short, for password ssh authentication it is 100% okay, but for ssh key authentication this actual version `0.4.0` is not compatible with any Paramiko version other than **`2.7.2`**

### Error messages:
* **Could not open connection to gateway**
*  **Could not establish session to SSH gateway**

### Actual Solution:
`pip install paramiko==2.7.2`

I like the fact that sshtunnel by default uses keys under ~/.ssh so no need to provide the ssh key param in most cases.
